core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/SegmentIndexFileStore.java

```diff
@@ -79,6 +79,11 @@
    */
   private Map<String, List<String>> carbonMergeFileToIndexFilesMap;
 
+  /**
+   * Stores the list of invalid index files of the SI segments in case of small files.
+   */
+  private static Set<String> oldSIIndexAndMergeIndexFiles = new HashSet<>();
```

Comment:
This would fail in case of concurrent queries as 1 operation may override for another

Reply:
removed the usage of oldSIIndexAndMergeIndexFiles , as #3999 deals with merge and write segment file based on UUID (timestamp), the list is no longer needed for these steps.